.storybook/main.js

```diff
@@ -1,4 +1,71 @@
+const path = require('path');
+
 module.exports = {
-  "stories": ["../src/**/*.stories.@(tsx)"],
-   "addons": ["@storybook/addon-essentials"]
-}
+	stories: ['../src/**/*.stories.@(tsx)'],
+	addons: ['@storybook/addon-essentials'],
+	webpackFinal: async (config) => {
+		const rules = config.module.rules;
+		const { extensions } = config.resolve;
+
+		// Support typescript in Storybook
+		// https://storybook.js.org/docs/configurations/typescript-config/
+		rules.push({
+			test: /\.[jt]sx?|mjs$/,
+			exclude: [
+				{
+					test: /node_modules/,
+					exclude: [
+						/@guardian\/(?!(automat-modules))/,
+						/dynamic-import-polyfill/,
+					],
+				},
+			],
+			use: [
+				{
+					loader: 'babel-loader',
+					options: {
+						presets: [
+							'@babel/preset-react',
+							[
+								'@babel/preset-env',
+								{
+									bugfixes: true,
+									targets: {
+										esmodules: true,
+									},
+								},
+							],
+						],
+					},
+				},
+				{
+					loader: 'ts-loader',
+					options: {
+						configFile: 'tsconfig.build.json',
+						transpileOnly: true,
+					},
```

Comment:
Bring these in from our babel loader as a replacement for awesome-typescript-loader